apps/src/dance/Dance.js

```diff
@@ -148,6 +148,17 @@ Dance.prototype.init = function(config) {
 
   this.initSongsPromise = this.initSongs(config);
 
+  // As students add new characters to their programs,
+  // load sprites asynchronously. Previously, we waited to load
+  // sprites once students hit the run button.
+  // This is in place to support iOS, which will not play audio
+  // if started asynchronously after fetching sprites.
+  const computeCharactersReferenced = () =>
```

Comment:
I was a bit confused about why this is declared as a function, rather than just getting the computed characters here. To confirm my understanding, we declare `computeCharactersReferenced` as a function, so that it is called each time we hit the changeHandler and recomputes the set of characters referenced. Does that sound accurate?

Reply:
Yup exactly!